### DISCUSSION TITLE
Transaction PriceLimit is only respected in the txpool (discussion)

### DISCUSSION THREAD

**Discussion by mrwillis** [0xPolygon/polygon-edge#281] [Q&A] [answered]
### Description
More of a discussion point, but curious why only the pool respects the PriceLimit and not nodes receiving transactions directly. 

server --help returns:

```
--price-limit PRICE_LIMIT
        Sets minimum gas price limit to enforce for acceptance into the pool. Default: 1
```

For example, if you had a four node cluster, you could submit a transaction with price of 0 to node 1, which it wouldn't propogate to nodes 2,3,4 because of this line.

https://github.com/0xPolygon/polygon-sdk/blob/develop/txpool/txpool.go#L512

Would it not make more sense to discard the transaction on node 1 as well, and reject at the RPC level?

### Your environment

* OS and version Ubuntu 20
* version of the Polygon SDK 6efebc6a92f885082620ed2a8d59e37d4d27074c
* branch that causes this issue `develop`

**Comment by Kourin1996** [comment]
@mrwillis Hi, if you want to make node1 discard the tx with lower price. You can give `--nolocals` flag when you start server.

**Comment by mrwillis** [comment]
Thanks @Kourin1996. Sorry documentation is a little confusing here. So If I set `--nolocals` on Node 1 that means the node won't accept any transactions lower than the gas limit into Node 1's pool?

**Comment by Kourin1996** [comment]
@mrwillis Node can accept transactions from JSON-RPC/gRPC even though gas price is less than price limit. Only remote transactions, which come from gossip broadcasted transactions, are restricted by price limit as default. If you give `--nolocals`, then node checks gas price for local transactions, also.

But @zivkovicmilos opened new PR https://github.com/0xPolygon/polygon-sdk/pull/244 and merged. It set default price limit to 0. So node will accept all transactions even though its gas price is 0 when you don't give `--price-limit`, `--nolocals`

**Comment by zivkovicmilos** [comment]
Hey @mrwillis,

I'll open up a PR that gives a better understanding of how `price-limit` works, and how it relates to `locals`.
Pinging you here once it's up 👍

**Comment by zivkovicmilos** [comment, marked as answer]
Hey @mrwillis,

I've added additional docs for the `price-limit` flag. They should be merged out soon 👍 
https://github.com/0xPolygon/polygon-sdk-docs/pull/34
